### PR TITLE
generic email action function simple implementation

### PR DESCRIPTION
**Pull request by Damilare-alesh99**
This branch just creates all email order templates and creates generic send email function where all emails can potentially be  sent through.

### CODE REVIEW

File: src/components/emails/orders/order-placed.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const OrderPlaced = () => {
+    return (
+      <div>
+        <p>Hi there,</p>
+        <p>
+        We’ve received your order. Please complete payment to continue.
```
Comment (KamsiyonnaObi) [inline src/components/emails/orders/order-placed.tsx]: The linter is not happy with your apostrophe `'` It prefers you use `&apos;`
Screenshot 2025-10-12 at 1 22 37 PM

File: src/components/emails/orders/order-payment-received.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const OrderPaymentReceived = () => {
+    return (
+      <div>
+        <p>Hi there,</p>
+        <p>
+         Your payment has been received. We're processing your order.
```
Comment (KamsiyonnaObi) [inline src/components/emails/orders/order-payment-received.tsx]: Same Linting error here. Maybe do a quick scan of all the templates you have created for apostrophes then replace with `&apos;`

File: src/components/emails/orders/order-confirmation.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    customerName: string;
+    orderDate: string;
+    shippingAddress: {
+      street: string;
+      city: string;
+      state: string;
+      zipCode: string;
+    };
+  };
+}
```
Comment (KamsiyonnaObi) [inline src/components/emails/orders/order-confirmation.tsx]: Are we using the `OrderConfirmationProps` ?

File: src/components/emails/orders/pickup-order-ready.tsx
Comment (KamsiyonnaObi) [inline src/components/emails/orders/pickup-order-ready.tsx:1]: We can rename the component to PickupOrderReady.

File: src/components/emails/orders/pickup-order-ready.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export const OrderReady = () => {
+    return (
+      <div>
+        <p>Hi there,</p>
+        <p>
+         Your order is ready for pickup at location.
+        </p>
```
Comment (KamsiyonnaObi) [inline src/components/emails/orders/pickup-order-ready.tsx:7]: Let the component ask for some props. We should show some details of what they need to pick up or just link them to their order confirmation page. 

Here's a sample Order Ready for pickup from Best Buy.

Screenshot 2025-10-12 at 1 38 33 PM

File: src/components/emails/orders/order-delivered.tsx
Comment (KamsiyonnaObi) [inline src/components/emails/orders/order-delivered.tsx:1]: Interestingly, Best Buy doesn't send an order delivered email. Instead, they attach an order status tracker to the order shipped email. We don't have to implement everything Best Buy has, but I am curious to hear your thoughts.

Screenshot 2025-10-12 at 1 44 45 PM

File: src/components/emails/orders/order-shipped.tsx
Comment (KamsiyonnaObi) [inline src/components/emails/orders/order-shipped.tsx:1]: `trackingID` might be a fluke. Sorry for misleading you there. It might be worth linking the user to their order confirmed page. 

Review my comments in the order-delivered component and let me know what you think

File: src/hooks/checkout/useCheckoutForm.ts [hunk 5]
```diff
@@ -95,9 +96,19 @@ export const useCheckoutForm = () => {
...
       if (!customerOrder?.createdOrder?.id) {
         throw new Error("Failed to create order");
       }
+
+         await sendEmail({
+        to: values.email,
+          subject: "Order Placed - Graceland",
+          template: "order-placed",
+          data: { }
+      })
```
Comment (KamsiyonnaObi) [inline src/hooks/checkout/useCheckoutForm.ts:111]: With this users will get emails before they attempt payments

Seems like users will have emails from abandoned orders.

Let's send order-placed only after payment is confirmed. thoughts?

File: src/hooks/checkout/usePaystack.ts [hunk 6]
```diff
@@ -35,6 +36,13 @@ export const usePaystack = (config: PaystackConfig) => {
           duration: 2000,
         });
         setTransactionReference(reference.trxref);
+
+         sendEmail({
+          to: config.email,
+          subject: "Order Payment Received - Graceland",
+          template: "order-payment-received",
+          data: { }
+      })
```
Comment (KamsiyonnaObi) [inline src/hooks/checkout/usePaystack.ts]: ...review comment from `...src/hooks/checkout/useCheckout.ts`

The payment received event is captured in our webhook. 

This should be moved to `src/app/api/paystack/webhook/route.ts`

File: src/server/actions/notifications.actions.ts
Comment (KamsiyonnaObi) [inline src/server/actions/notifications.actions.ts:1]: What happens when an invalid string case is passed for template? You can ask copilot to generate unit test for this function and it should provide edge cases for us to consider. 

We don't have to push the unit tests for now, but just for copilot to catch any bugs we might have missed.

**Review by KamsiyonnaObi — CHANGES_REQUESTED** [review]

File: src/components/emails/orders/order-confirmation.tsx (hunk 3 above)
Comment (Damilare-alesh99) [reply src/components/emails/orders/order-confirmation.tsx]: not as of now, will remove

File: src/components/emails/orders/order-delivered.tsx
Comment (Damilare-alesh99) [reply src/components/emails/orders/order-delivered.tsx:1]: Will probably need to discuss more about this but I think for now especially a delivered email won't hurt as we have no other tracking I think.

File: src/hooks/checkout/useCheckoutForm.ts (hunk 5 above)
Comment (Damilare-alesh99) [reply src/hooks/checkout/useCheckoutForm.ts:111]: sure thing, it should only be sent after payment too. In that case then we'll have one less email because right now we have order placed and order confirmation email, thay can be merged.

File: src/components/emails/orders/order-delivered.tsx
Comment (KamsiyonnaObi) [reply src/components/emails/orders/order-delivered.tsx:1]: Sure, we can finalize when we talk on Friday.

File: src/app/api/paystack/webhook/route.ts [hunk 7]
```diff
@@ -28,6 +28,14 @@ export async function POST(req: NextRequest) {
     if (event === "charge.success") {
       await updateOrderStatusAndSavePaymentInfo(data);
 
+
+      sendEmail({
```
Comment (KamsiyonnaObi) [inline src/app/api/paystack/webhook/route.ts]: there is no `await` keyword. I guess there is no way to test if the email actually goes through.

File: src/components/emails/orders/order-confirmation.tsx
Comment (KamsiyonnaObi) [inline src/components/emails/orders/order-confirmation.tsx:1]: To be discussed on Friday. Is it worth while to include order details in the emails now or is that beyond the scope of this task.
